src/main/java/seedu/toluist/commons/util/DateTimeFormatterUtil.java

```diff
@@ -40,6 +58,99 @@ public static String formatEventRange(LocalDateTime from, LocalDateTime to) {
         }
     }
 
+    /**
+     * Format recurring task with deadline
+     */
+    public static String formatRecurringTaskDeadline(LocalDateTime deadline, LocalDateTime recurringEndDateTime,
+            RecurringFrequency recurringFrequency) {
+        String formattedResult = "";
+        switch (recurringFrequency) {
+        case DAILY:
+            formattedResult += String.join(" ", EVERY, formatTime(deadline), OF_THE, DAY);
+            break;
+        case WEEKLY:
+            formattedResult += String.join(" ", EVERY, formatDayOfWeek(deadline), OF_THE, WEEK);
+            break;
+        case MONTHLY:
+            formattedResult += String.join(" ", EVERY, formatDayOfMonth(deadline), OF_THE, MONTH);
+            break;
+        case YEARLY:
+            formattedResult += String.join(" ", EVERY, formatDateOfYear(deadline), OF_THE, YEAR);
+            break;
+        default:
+            // won't happen
+        }
+        if (recurringEndDateTime != null) {
+            formattedResult += RECURRING_DATE_SEPARATOR + String.join(" ", UNTIL, formatDate(recurringEndDateTime));
```

Comment:
In hindsight, maybe i can just put recurring date separator to replace all my " " magic string